cg/meta/store/base.py

```diff
@@ -1,7 +1,58 @@
 """ Base module for building bioinfo workflow bundles for linking in Housekeeper"""
 import datetime as dt
+import logging
+from pathlib import Path
 
-from cg.exc import AnalysisDuplicationError, PipelineUnknownError
+import ruamel.yaml
+
+from cg.constants import TAGS
+from cg.exc import (
+    AnalysisDuplicationError,
+    PipelineUnknownError,
+    BundleAlreadyAddedError,
+    AnalysisNotFinishedError,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+def gather_files_and_bundle_in_housekeeper(config_stream, hk_api, status):
+    """Function to gather files and bundle in housekeeper"""
+    bundle_data = add_analysis(config_stream)
+
+    results = hk_api.add_bundle(bundle_data)
+    if results is None:
+        raise BundleAlreadyAddedError("bundle already added")
+
+    bundle_obj, version_obj = results
+
+    case_obj = status.family(bundle_obj.name)
+
+    reset_case_action(case_obj)
+    new_analysis = add_new_analysis(bundle_data, case_obj, status, version_obj)
+    version_date = version_obj.created_at.date()
+
+    LOG.info("new bundle added: %s, version %s", bundle_obj.name, version_date)
+    hk_api.include(version_obj)
+    hk_api.add_commit(bundle_obj, version_obj)
+
+    return new_analysis
+
+
+def add_analysis(config_stream):
```

Comment:
```suggestion
def add_mip_analysis(config_stream):
```
There will most likely be a balsamic `add_balsamic_analysis` soon.

Reply:
Or you should make a store/mip.py and define all MIP specific stuff there and import where relevant

Reply:
Created mip.py and moved add_analysis into it. https://github.com/Clinical-Genomics/cg/pull/621/commits/c2c319ce6125724bfc93f65c3d6bb4f36ae5a80f